charts/opentelemetry-collector/values.yaml

```diff
@@ -256,6 +256,9 @@ hostNetwork: false
 # Pod DNS policy ClusterFirst, ClusterFirstWithHostNet, None, Default, None
 dnsPolicy: ""
 
+# Custom DNS config. Required when DNS policy is None.
```

Comment:
Let's validate this requirement in the `NOTES.txt` with a failure.  If `dnsPolicy=none` and `dnsConfig` is empty lets fail the install.